docs/configuring-entities.md

```diff
@@ -0,0 +1,197 @@
+```
+
+### MkDocs
+
+TechDocs plugin uses [MkDocs][12] as the builder. Your directory structure should look like the one bellow. For more information on how to setup MkDocs, follow [this guide][14].
+
+```
+docs/
+├── index.md
+mkdocs.yaml
+```
+
+This default setup above is the most common use case when you have one documentation per one entity repository. However it is also posible to have multiple documentations for multiple entities in one repository. Each documented entity has to be contained in one directory. See the example below:
+
+```
+service1/
+├── docs/
+│   └── index.md
+├── mkdocs.yaml
+docs/
+├── index.md
+mkdocs.yaml
+```
+
+Depending on which setup you will chose the configuration of the workflow will be different.
+
+### Using TechDocs workflow template
+
+A template can be found [here][9], after the template has been coppiced uncomment this part:
+
+```yaml
+on:
+  push:
+    branches:
+      - main
+      - master
+    paths:
+      - "**/docs/**"
+      - "mkdocs.yaml"
+      - "mkdocs.yml"
+```
+
+since it will allow the workflow to run on any changes made to the documentation.
+
+Next up define the steps for the workflow. Each step has to reference [this][11] action which is used to activate the build and publish process. A code snipped below explains how to use this action when your documentation is placed in the root of your directory:
+
+```yaml
+- name: Send dispatch for service-catalog
+  uses: operate-first/service-catalog/.github/actions/docs-dispatch@main
+  with:
+    # Root of the repository
+    docs_path: '.'
+    # Name of your entity in the catalog
+    entity_name: 'service-catalog'
+    # Type of your entity in the catalog
+    entity_kind: 'Component'
+    # This can always be ${{ github.repository }}
+    repository: ${{ github.repository }}
+    # GitHub token that requires metadata:read and contents:read&write permissions to service-catalog repository
+    token: ${{ secrets.SESHETA_TOKEN }}
+```
+
+If the documentation is not located in the root of the repository, a filter definition is required using [glob expresions][15]:
+
+```yaml
+- name: Send dispatch for service2
+  uses: operate-first/service-catalog/.github/actions/docs-dispatch@main
+  with:
+    # Glob expression, filtering where your documentation root is
+    filter: 'service2/**'
+    # Directory path to your documentation folder relative to the repository root
+    docs_path: 'service2'
+    entity_name: 'service2'
+    entity_kind: 'Component'
+    repository: ${{ github.repository }}
+    token: ${{ secrets.SESHETA_TOKEN }}
+```
+
+<!--Mermaid diagram showing techdocs architecture somewhere here -->
+[Documentation reference][13]
+
+## GitHub Insights/Security Insights
+
+For the insights plugins to work it is required to put `github.com/project-slug` annotation to your components definition. The value of this annotation is a GitHub project slug which consists of the organization/user and the repository name, see the example below.
```

Comment:
Missing a description/short intro about the plugin.

I don't think we need to explain the value format, if we show an example.. it's quite self explanatory...

```suggestion
These GitHub specific plugins can display various GitHub content to the user. GitHub Insights renders a readme, contributors and such. Security Insights can display pending security advisories. Both plugins rely on a `github.com/project-slug` annotation presence in the entity definition.
```

Reply:
Resolved in the latest commit.